src/concurrency/timestamp_ordering_transaction_manager.cpp

```diff
@@ -778,130 +778,136 @@ ResultType TimestampOrderingTransactionManager::CommitTransaction(
   oid_t database_id = 0;
   if (static_cast<StatsType>(settings::SettingsManager::GetInt(settings::SettingId::stats_mode)) !=
       StatsType::INVALID) {
-    if (!rw_set.empty()) {
-      database_id =
-          manager.GetTileGroup(rw_set.begin()->first)->GetDatabaseId();
+    if (!rw_set.IsEmpty()) {
+      // Call the GetIterator() function to explicitly lock the cuckoohash
+      // and initilaize the iterator
+      auto rw_set_lt = rw_set.GetConstIterator();
+      const auto tile_group_id = rw_set_lt.begin()->first.block;
+      database_id = manager.GetTileGroup(tile_group_id)->GetDatabaseId();
     }
   }
 
   // install everything.
   // 1. install a new version for update operations;
   // 2. install an empty version for delete operations;
   // 3. install a new tuple for insert operations.
-  for (auto &tile_group_entry : rw_set) {
-    oid_t tile_group_id = tile_group_entry.first;
-    auto tile_group_header = manager.GetTileGroup(tile_group_id)->GetHeader();
+  // Iterate through each item pointer in the read write set
+
+  // TODO: This might be inefficient since we will have to get the
```

Comment:
As discussed in the cleanup week, we may want to add the name after the TODO so that others would know who to refer to if they want to later improve this part of the code.